### PR TITLE
luminous: auth: ceph auth add does not sanity-check caps

### PR DESCRIPTION
**Pull request by tchaikov**
http://tracker.ceph.com/issues/23670

### CODE REVIEW

File: PendingReleaseNotes [hunk 1]
```diff
@@ -111,3 +111,7 @@
     has an error.  This head object is there to show the snapset that was used in
     determining errors.
 
+* In 10.2.10 and earlier releases, keyring caps were not checked for validity,
+  so the caps string could be anything. As of 10.2.11, caps strings are
+  validated and providing a keyring with an invalid caps string to, e.g.,
+  "ceph auth add" will result in an error.
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply PendingReleaseNotes]: @smithfarm thank you! i was blindly cherry-picking the release notes, should have reviewed it first =( 

fixed and repushed.

[the remark this answers is not in the document]
Comment (smithfarm) [reply PendingReleaseNotes]: @tchaikov Thanks for the backport.

**Comment by smithfarm** [comment]
@tchaikov Please rebase

**Comment by yuriw** [comment]
@tchaikov pls rebase

--- pr 24906 --- pulling https://github.com/tchaikov/ceph.git branch wip-23673-luminous
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Total 21 (delta 16), reused 16 (delta 16), pack-reused 5
Unpacking objects: 100% (21/21), done.
From https://github.com/tchaikov/ceph
 * branch            wip-23673-luminous -> FETCH_HEAD
Auto-merging PendingReleaseNotes
CONFLICT (content): Merge conflict in PendingReleaseNotes
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r
AssertionError

**Comment by tchaikov** [comment]
@smithfarm @yuriw rebased.

**Comment by yuriw** [comment]
@tchaikov still has conflict: 
--- pr 24906 --- pulling https://github.com/tchaikov/ceph.git branch wip-23673-luminous
remote: Enumerating objects: 16, done.
remote: Counting objects: 100% (16/16), done.
remote: Total 21 (delta 16), reused 16 (delta 16), pack-reused 5
Unpacking objects: 100% (21/21), done.
From https://github.com/tchaikov/ceph
 * branch            wip-23673-luminous -> FETCH_HEAD
Auto-merging PendingReleaseNotes
CONFLICT (content): Merge conflict in PendingReleaseNotes
Automatic merge failed; fix conflicts and then commit the result.
Traceback (most recent call last):
  File "/home/yuriw/wip_master/src/script/build-integration-branch", line 62, in <module>
    assert not r

**Comment by tchaikov** [comment]
@yuriw rebased and repushed

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-02-11-1725-luminous](https://trello.com/c/82sp4VSF/535-wip-yuri4-testing-2019-02-11-1725-luminous)

**Comment by smithfarm** [comment]
@tchaikov This was merged after the v12.2.11 release and before v12.2.12 so the release notes commit is wrong, I believe. Instead of "In 12.2.9 and earlier releases, keyring caps were not checked for validity, so the caps string could be anything. As of 12.2.10, ..." it should say

"In 12.2.11 and earlier releases, . . . . As of 12.2.12, . . . ."

Is that correct?

**Comment by tchaikov** [comment]
@smithfarm correct =)

**Pull request merged**